donation_base/models/donation_tax_receipt.py

```diff
@@ -0,0 +1,99 @@
+# © 2014-2016 Barroux Abbey (http://www.barroux.org)
+# © 2014-2016 Akretion France (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class DonationTaxReceipt(models.Model):
+    _name = "donation.tax.receipt"
+    _inherit = ["mail.thread"]
+    _description = "Tax Receipt for Donations"
+    _order = "id desc"
+    _rec_name = "number"
+
+    number = fields.Char(string="Receipt Number")
+    date = fields.Date(
+        string="Date", required=True, default=fields.Date.context_today, index=True
+    )
+    donation_date = fields.Date(string="Donation Date")
+    amount = fields.Monetary(string="Amount", currency_field="currency_id")
+    currency_id = fields.Many2one(
+        "res.currency",
+        string="Currency",
+        required=True,
+        ondelete="restrict",
+        default=lambda self: self.env.company.currency_id.id,
```

Comment:
why was this change?
before:
`self.env.user.company_id.currency_id.id`

Reply:
On 13 this is the way to get the current company

Reply:
no it's still
`self.env.user.company_id.currency_id`
for multicompany you need the company_id of the user, which is the previous line

`self.env.company`
just gets the current env company

Reply:
Not according the migration guide
https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0
https://www.odoo.com/documentation/13.0/developer/howtos/company.html

Reply:
The inconsistency of odoo :)
https://github.com/odoo/odoo/blob/13.0/odoo/addons/base/models/res_company.py#L194

they still using `self.env.user.company_id` themself.
as both are working I would preffer to use the previous one, as it makes it easier to find for possible patches and fixes later on as the code stays consistent

Reply:
In your comment it is using user.company_ids, it is not the same

Reply:
https://github.com/odoo/odoo/blob/13.0/addons/repair/models/repair.py#L312

Reply:
In any case, we should must env.company_id, as this one is the specified in the context, otherwise, it will use env.user.company_id.

https://github.com/odoo/odoo/blob/13.0/odoo/api.py#L525

allowed_company_ids is defined in the selector of the company. On the previous Odoo versions, multicompany was solved using the parent_id. Now it is solved using a multiselector on the header. So, this should be the way.

Reply:
I confirm that we must use self.env.company in v13 and not self.env.user.company_id. Obviously, the code of the official addons is far from perfect and Odoo itself hasn't achieved the full migration !